packages/components/addon/components/hds/dropdown/index.hbs

```diff
@@ -0,0 +1,38 @@
+<Hds::Disclosure class="hds-dropdown">
+  <:toggle as |t|>
+    {{#if (eq @toggle "toggleButton")}}
```

Comment:
For context and future reference: [thread here](https://hashicorp.slack.com/archives/C11JCBJTW/p1649413297164169).

Reply:
let's go ahead and change it. We can let teams try it out and if it feels confusing we can switch it back. WDYT?

Reply:
ok after reading more comments and seeing the thing about nesting yet another named block in here, I think we should leave this exactly like it is. It's quickly spilling over into over-engineered territory.

Reply:
I don't agree. Everyone that commented in the thread is saying the current implementation is not ergonomic. Why would you go with that solution?

Reply:
That's factually inaccurate. Let's schedule a call to discuss.

Reply:
OK, let me rephrase it. Considered these feedbacks:
- _Jamie White - "That third option is a little more verbose, but seems the clearest to me." + "I actually have no strong opinions on this API. Y’all are thinking about this really deeply and I trust y’all to come up with a balanced design. I would be happy with any of the APIs presented."_
- _John Cowen - "of those options I would prefer 2"_
- _Matthew Irish - "[option 3] is pretty familiar since it’s pretty close to ember-basic-dropdown which is what hcp’s current dropdown was patterned after"_
and considered these are the only feedback we have received, why would you go with option 1?

Reply:
After discussion with @MelSumner during code review, we decided to go with the yielded components solution.

Reply:
Done in #185